test/boost/sstable_3_x_test.cc

```diff
@@ -3200,55 +3203,59 @@ static sstring get_write_test_path(sstring table_name) {
 }
 
 // This method should not be called for compressed sstables because compression is not deterministic
-static void compare_sstables(const std::filesystem::path& result_path, sstring table_name, sstable_version_types version) {
+static void compare_sstables(const std::filesystem::path& result_path, sstring table_name, sstable_version_types version, sstables::shared_sstable sst) {
     for (auto file_type : {component_type::Data,
                            component_type::Index,
                            component_type::Digest,
                            component_type::Filter}) {
         auto orig_filename =
                 sstable::filename(get_write_test_path(table_name),
-                                  "ks", table_name, sstables::sstable_version_types::mc, generation_from_value(1), big, file_type);
+                                  "ks", table_name, sstable_version_types::mc, generation_from_value(1), big, file_type);
         auto result_filename =
-                sstable::filename(result_path.string(), "ks", table_name, version, generation_from_value(1), big, file_type);
+                sstable::filename(result_path.string(), "ks", table_name, sst->get_version(), sst->generation(), big, file_type);
```

Comment:
This change doesn't fit the "use make_sstable defaults as much as possible" description

Reply:
extracted out in next version (need actually a few patch to make incremental, standalone changes)